### PR TITLE
defect2278780

### PR DESCRIPTION
**Pull request by MokireddySampath**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1472?feature.someFeatureFlagYouMightNeed=true)

defect- https://msdata.visualstudio.com/CosmosDB/_workitems/edit/2278780

Rows present under the table in Entities are not accessible through keyboard. Keyboard navigation has been fixed for the rows of the table in "Entities" tab.

### CODE REVIEW

**Review by vchske — APPROVED** [review]

File: src/Explorer/Tables/DataTable/DataTableBindingManager.ts [hunk 1]
```diff
@@ -210,6 +213,39 @@ function selectionChanged(element: any, valueAccessor: any, allBindings: any, vi
     });
   //selected = bindingContext.$data.selected();
 }
+function handlearrowkey(
```
Comment (jawelton74) [inline src/Explorer/Tables/DataTable/DataTableBindingManager.ts:216]: Camel casing on the function name please

**Review by asierisayas — APPROVED** [review]

**Review by sindhuba — APPROVED** [review]

File: src/Explorer/Tables/DataTable/DataTableBindingManager.ts [hunk 2]
```diff
@@ -208,6 +213,39 @@ function selectionChanged(element: any, valueAccessor: any, allBindings: any, vi
...
 }
+function handleArrowKey(
+  element: any,
+  valueAccessor: any,
+  allBindings: any,
+  viewModel: any,
+  bindingContext: any,
+  event: JQueryEventObject,
+) {
+  let isUpArrowKey: boolean = event.keyCode === Constants.keyCodes.UpArrow;
```
Comment (vchske) [inline src/Explorer/Tables/DataTable/DataTableBindingManager.ts:224]: Nit: if values never change, they should be marked as "const" not "let".

File: src/Explorer/Tables/DataTable/DataTableBindingManager.ts [hunk 3]
```diff
@@ -208,6 +213,39 @@ function selectionChanged(element: any, valueAccessor: any, allBindings: any, vi
...
+  let isDownArrowKey: boolean = event.keyCode === Constants.keyCodes.DownArrow;
+
+  if (isUpArrowKey || isDownArrowKey) {
+    let $dataTable = $(element);
```
Comment (vchske) [inline src/Explorer/Tables/DataTable/DataTableBindingManager.ts:228]: Same here

File: src/Explorer/Tables/DataTable/DataTableBindingManager.ts [hunk 4]
```diff
@@ -208,6 +213,39 @@ function selectionChanged(element: any, valueAccessor: any, allBindings: any, vi
...
+    let $selectedRow = $dataTable.find("tr.selected");
+
+    if ($selectedRow.length === 0) {
+      // No row is currently selected, select the first row
+      $selectedRow = $dataTable.find("tr:first");
+      $selectedRow.addClass("selected");
+    } else {
+      let $targetRow = isUpArrowKey ? $selectedRow.prev("tr") : $selectedRow.next("tr");
```
Comment (vchske) [inline src/Explorer/Tables/DataTable/DataTableBindingManager.ts:236]: Same here

**Review by vchske — APPROVED** [review]

**Review by vchske — APPROVED** [review]

**Review by jawelton74 — APPROVED** [review]

**Pull request merged**